widget/tree.go

```diff
@@ -186,6 +188,40 @@ func (t *Tree) Selection() string {
 // SetSelection updates the current selection to the node with the given Unique ID.
 func (t *Tree) SetSelection(uid string) {
 	t.selected = uid
+	if t.selected != TreeNoSelection && t.scroller != nil {
+		var found bool
+		var y int
+		var size fyne.Size
+		t.walkAll(func(uid string, branch bool, depth int) {
+			m := t.leafMinSize
+			if branch {
+				m = t.branchMinSize
+			}
+			if uid == t.selected {
+				found = true
+				size = m
+			} else if !found {
+				// Root node is not rendered unless it has been customized
+				if t.Root == "" && uid == "" {
+					// This is root node, skip
+					return
+				}
+				// If this is not the first item, add a divider
+				if y > 0 {
+					y += treeDividerHeight
+				}
+
+				y += m.Height
+			}
+		})
```

Comment:
```suggestion
				size = m
				return
			}
			// Root node is not rendered unless it has been customized
			if t.Root == "" && uid == "" {
				// This is root node, skip
				return
			}
			// If this is not the first item, add a divider
			if y > 0 {
				y += treeDividerHeight
			}
			y += m.Height
		}
	})
```

Reply:
Returning early does not stop walkAll from iterating through the tree

Reply:
I get that - but it makes for cleaner code here does it not?  Especially if it's going to iterate regardless?